File: lib/base16/lib/base16.js
var MAPS = require('./../../maps'),
    MAP16 = MAPS.arrs[16];


module.exports = {

  /**
   * @scope exports object
   * @param b16 {string}
   * @return b2 {string}
   */
  '2': function (){
    return function (b16){
      var b2 = '',
          b10 = this[10](b16),
          rev_b16_arr = b16.toLowerCase().split('').reverse();

      while(b10 != 0){
        b2 = (b10 % 2) + b2;
        b10 = Math.floor(b10 / 2);
      }

      return b2;
    };
  }(),
  
  /**
   * @param b16  {string}
   * @return b10 {int}
   */
  '10': function (b16){
    var b10 = 0,
        rev_b16_arr = b16.toLowerCase().split('').reverse();

    for(var i=0; i<rev_b16_arr.length; i+=1)
      b10 += MAP16.indexOf(rev_b16_arr[i]) * Math.pow(16, i);
    
    return b10;
  }
  
};

File: lib/base16/cpp/base16.cc
#include <nan.h>

#include <iostream>
#include <string>
#include <sstream>
#include <stdio.h>
#include <math.h>

#include "maps.h"

using namespace v8;
using namespace std;

NAN_METHOD(B2){
  NanScope();

  cout << "b2" << endl;

  NanReturnValue(NanNew("ok"));
}

NAN_METHOD(B10){
  NanScope();

  cout << "b10" << endl;

  NanReturnValue(NanNew("ok"));
}


void Init(Handle<Object> exports) {
  exports->Set(NanNew("2"), NanNew<FunctionTemplate>(B2)->GetFunction());
  exports->Set(NanNew("10"), NanNew<FunctionTemplate>(B10)->GetFunction());
}

NODE_MODULE(b16, Init)